src/curve_fitting.py

```diff
@@ -0,0 +1,272 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import re
+from gettext import gettext as _
+
+from gi.repository import Adw, GObject, Graphs, Gtk
+
+from graphs import ui
+from graphs.canvas import Canvas
+from graphs.data import Data
+from graphs.item import Item
+from graphs.utilities import preprocess, string_to_function
+
+import numpy
+
+from scipy.optimize import curve_fit
+
+
+@Gtk.Template(resource_path="/se/sjoerd/Graphs/ui/curve_fitting.ui")
+class CurveFittingWindow(Adw.Window):
+    __gtype_name__ = "CurveFittingWindow"
+    equation_entry = Gtk.Template.Child()
+    fitting_params = Gtk.Template.Child()
+    toast_overlay = Gtk.Template.Child()
+    text_view = Gtk.Template.Child()
+
+    def __init__(self, application, item):
+        super().__init__(
+            application=application, transient_for=application.get_window(),
+
+        )
+        ui.bind_values_to_settings(
+            self.get_application().get_settings("curve-fitting"), self)
+        canvas = Canvas(application)
+        self.param = []
+
+        # Set data item that contain the curves
+        self.curves = Data(application)
+        self.equation_entry.connect("notify::text", self.on_equation_change)
+        self.fitting_parameters = FittingParameters(application)
+
+        for var in self.get_free_variables():
+            self.fitting_parameters.add_items([FittingParameter(var, 1)])
+
+        # Generate item for the data that is fitted to
+        data_curve = Item.new(
+            application, xdata=item.xdata,
+            ydata=item.ydata, name=item.name,
+            color="#1A5FB4")
+        data_curve.linestyle = 0
+        data_curve.markerstyle = 1
+        data_curve.markersize = 13
+
+        # Generate item for the fit
+        fitted_curve = Item.new(
+            application, xdata=[], ydata=[], color="#A51D2D")
+
+        self.curves.add_unconnected_items([fitted_curve, data_curve])
+        self.fit_curve()
+        self.set_entry_rows()
+
+        # Set figure settings
+        figure_settings = \
+            Graphs.FigureSettings.new(
+                application.get_settings().get_child("figure"))
+        for prop in dir(figure_settings.props):
+            if prop not in ["use_custom_style", "custom_style"]:
+                figure_settings.bind_property(prop, canvas, prop, 1 | 2)
```

Comment:
The figure settings aren't used any further any reason to create them?

Reply:
This was left-over code in an attempt  to couple the figure style to the main canvas without coupling the other settings. The styles are carried over properly when initiating the window, but when changing style from dark to light and vice verse, the plot style is still kept in the original mode. This is something that still needs to be added to this code.